promiseWrapper.js

```diff
@@ -1,138 +1,77 @@
-var credentials = require('./DONOTCOMMIT.js')();
-var options = {
-  host: credentials.TESTHost(),
-  username: credentials.TESTUser(),
-  password: credentials.TESTPass(),
-  port: credentials.TESTPort()
-};
 
-// var connect = (options) => {
-
-//   return new Promise((resolve, reject) => {
-//     var SSHClient = require('./lib/index.js')(options);
-//     SSHClient.on('stderr', function(err) { console.log("Error on StdErr",err); });
-//     SSHClient.on('output', function(data,lastChunk) { 
-// 	    if (lastChunk == true) {
-// 		    console.log("Should close connection");
-// 		    SSHClient.close();
-// 	    }
-// 	    if (data) {
-// 		    if (data.match("pwd")) { console.log("Command data->",data);
-// 		                           }else { console.log("Output data->",data); }
-// 	    }
-//     });
-//     //Connection
-//     SSHClient.once('closed', function(addr,err) { 
-// 	    if (err) {
-// 		    console.log("Error on session",err); 
-// 	    }else { console.log("SSH Session has being closed."); }
-//     });
-//     SSHClient.on('connected',(host) => {
-//       console.log('Were Connected',host);
-//       SSHClient.write('pwd');
-//       resolve()
-//     });
-//     SSHClient.connect();
-//     //  resolve('uh connected I think')
-//   });
-// };
-
-// connect(options).then((string) => {
-//   console.log('amazing');
-//   console.log(string);
-// });
+'use strict';
 
+const shellEscape = require('shell-escape');
 
 class SSH {
+  // new SSH(options: host, username, password): Promise<this>
   constructor(options) {
+    this.connected = false;
     this.connection = null;
     this.options = options;
     this.connection = require('./lib/index.js')(this.options);
-    this.connection.on('stderr', function(err) { console.log("Error on StdErr",err); });
-    this.connection.once('closed', function(addr,err) {
-	    if (err) {
-		    console.log("Error on session",err);
-	    }else { console.log("SSH Session has being closed."); }
-    });
+    this.connection.on('stderr', function(err) { console.error('Error on StdErr',err); });
   }
+
+  //  connect(): Promise<this>
+
   connect() {
     return new Promise((resolve, reject) => {
       this.connection.on('closed', reject);
       this.connection.on('connected',(host) => {
         this.connection.removeListener('closed', reject);
-        console.log('Were Connected',host);
-        resolve();
+        this.connected = true;
+        resolve('Connected to host:', host);
       });
       this.connection.connect();
     });
   }
-  write(value) {
+
+
+  // execCommand(command: string, options: { cwd: string, stdin: string } = {}): Promise<{ stdout: string}>
+  execCommand(value, options) {
     return new Promise((resolve, reject) => {
-      console.log('in promise')
+      if (options.cwd) {
+        value = `cd ${shellEscape([options.cwd])} 1> /dev/null 2> /dev/null; ${value}`;
+      }
+      var wholeLine = '';
```

Comment:
var is deprecated. We should use let or const.

https://odetocode.com/blogs/scott/archive/2014/07/31/the-features-of-es6-part-1-let.aspx
https://odetocode.com/blogs/scott/archive/2014/08/06/features-of-es6-part-2-const.aspx

Reply:
ok cool